### ISSUE TITLE
Problems without specific revision of pipeline when pulling via Singularity

### ISSUE THREAD

**Issue by apeltzer** [SciLifeLab/NGI-RNAseq#209]
Hi!
neither on local nor on our cluster (binac) I can download the appropriate image... 
```
[warm up] executor > local
ERROR ~ Error executing process > 'fastqc (null)'

Caused by:
  Failed to pull singularity image
  command: singularity pull --name scilifelab-ngi-rnaseq-master.img docker://scilifelab/ngi-rnaseq:master > /dev/null
  status : 1
  message:
    WARNING: pull for Docker Hub is not guaranteed to produce the
    WARNING: same image on repeated pull. Use Singularity Registry
    WARNING: (shub://) to pull exactly equivalent images.
    ERROR MANIFEST_UNKNOWN: manifest unknown
    ERROR: pulling container failed!
-- Check '.nextflow.log' file for details
```

I think the problematic part is this code piece here in `base.config`:

```nextflow
container = { "scilifelab/ngi-rnaseq:${workflow.revision ? workflow.revision : 'latest'}" }
``` 

This defaults to something else, namely namely this command:

``` 
singularity pull --name scilifelab-ngi-rnaseq-master.img docker://scilifelab/ngi-rnaseq:master
```
Which doesn't work unfortunately - removing the master and things are fine ;-) 

I can send a pull request to fix this in case you'd be interested.

**Comment by ewels** [comment]
Ah crap - `workflow.revision` is resolving to `master`? I think this is a new change.. Pretty sure that used to be blank before if not a release tag.

_Thinking out loud:_ perhaps we can use `params.version` here (the version of the pipeline). If `-r v1.4` is used, the code executed with have the older `params.version` number and then that tag will be used for the container label. So maybe we should keep things simple and just have a static version number and container string which is updated for each release..

eg: development branch:

```nextflow
params.version = '1.5dev'
container = 'scilifelab/ngi-rnaseq'
```

Then bump for a release:


```nextflow
params.version = '1.5'
container = 'scilifelab/ngi-rnaseq:1.5'
```

**Comment by ewels** [comment]
This is the root cause behind the latest problem we've seen when trying to fix #200

**Comment by apeltzer** [comment]
I think your proposed solution could work - I'll test this quickly locally and send a pull request in a separate branch then :)
